### ISSUE TITLE
Assertion Failed: chunk.ColumnCount() == op.aggregates.size()

### ISSUE THREAD

**Issue by fuboat** [duckdb/duckdb#3349]
#### What happens?
Assertion triggered in file "/root/duckdb/src/execution/radix_partitioned_hashtable.cpp" on line 344: chunk.ColumnCount() == op.aggregates.size()

#### To Reproduce
```sql
create table strings (a VARCHAR, b VARCHAR);
select b, b, count(*) from strings group by rollup (b, b) order by 1, 2, 3;
```

#### Environment (please complete the following information):
 - OS: linux
 - DuckDB Version: v0.3.3-dev1395 80ae1e12d
 - DuckDB Client: /usr/local/bin/duckdb 

#### Before Submitting

- [x] **Have you tried this on the latest `master` branch?**
* **Python**: `pip install duckdb --upgrade --pre`
* **R**: `install.packages("https://github.com/duckdb/duckdb/releases/download/master-builds/duckdb_r_src.tar.gz", repos = NULL)`
* **Other Platforms**: You can find binaries [here](https://github.com/duckdb/duckdb/releases/tag/master-builds) or compile from source.

- [x] **Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?**

**Comment by fuboat** [comment]
ASAN detail:

```
AddressSanitizer:DEADLYSIGNAL
=================================================================
==26117==ERROR: AddressSanitizer: SEGV on unknown address 0x0000000000c8 (pc 0x000001919ae9 bp 0x60200000ed10 sp 0x7ffcba0d4ee0 T0)
==26117==The signal is caused by a READ memory access.
==26117==Hint: address points to the zero page.
    #0 0x1919ae9 in duckdb::RadixPartitionedHashTable::GetData(duckdb::ExecutionContext&, duckdb::DataChunk&, duckdb::GlobalSinkState&, duckdb::GlobalSourceState&) const /root/duckdb/src/execution/radix_partitioned_hashtable.cpp:350:54
    #1 0x269af16 in duckdb::PhysicalHashAggregate::GetData(duckdb::ExecutionContext&, duckdb::DataChunk&, duckdb::GlobalSourceState&, duckdb::LocalSourceState&) const /root/duckdb/src/execution/operator/aggregate/physical_hash_aggregate.cpp:260:34
    #2 0x5f586a in duckdb::PipelineExecutor::FetchFromSource(duckdb::DataChunk&) /root/duckdb/src/parallel/pipeline_executor.cpp:316:19
    #3 0x5ff172 in duckdb::PipelineExecutor::Execute(unsigned long) /root/duckdb/src/parallel/pipeline_executor.cpp:56:3
    #4 0x5ff172 in duckdb::PipelineTask::ExecuteTask(duckdb::TaskExecutionMode) /root/duckdb/src/parallel/pipeline.cpp:42:39
    #5 0x5e0934 in duckdb::ExecutorTask::Execute(duckdb::TaskExecutionMode) /root/duckdb/src/parallel/executor_task.cpp:17:10
    #6 0x5ed394 in duckdb::Executor::ExecuteTask() /root/duckdb/src/parallel/executor.cpp:350:24
    #7 0x5322d1 in duckdb::ClientContext::ExecuteTaskInternal(duckdb::ClientContextLock&, duckdb::PendingQueryResult&) /root/duckdb/src/main/client_context.cpp:336:41
    #8 0x533ea3 in duckdb::PendingQueryResult::ExecuteTaskInternal(duckdb::ClientContextLock&) /root/duckdb/src/main/pending_query_result.cpp:45:18
    #9 0x533ea3 in duckdb::PendingQueryResult::ExecuteInternal(duckdb::ClientContextLock&, bool) /root/duckdb/src/main/pending_query_result.cpp:50:9
    #10 0x518357 in duckdb::ClientContext::ExecutePendingQueryInternal(duckdb::ClientContextLock&, duckdb::PendingQueryResult&, bool) /root/duckdb/src/main/client_context.cpp:700:15
    #11 0x518357 in duckdb::ClientContext::Query(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool) /root/duckdb/src/main/client_context.cpp:639:21
    #12 0x519c06 in duckdb::Connection::Query(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) /root/duckdb/src/main/connection.cpp:71:25
    #13 0x4e57c8 in duckdb_query /root/duckdb/src/main/capi/duckdb-c.cpp:67:22
    #14 0x4e10d3 in run_one_statement_new(void*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) /mnt/jingzhou_workspace/sqlsim/client_new/duckdb/client_new.cpp:57:13
    #15 0x4d3984 in run_testcase(void*&, char const*, int) /root/autodriver.cpp:102:14
    #16 0x4d8601 in do_fuzz()::$_3::operator()() const /root/autodriver.cpp:227:13
    #17 0x4d809e in do_fuzz() /root/autodriver.cpp:235:13
    #18 0x4d8b43 in main /root/autodriver.cpp:292:9
    #19 0x7f96e9d510b2 in __libc_start_main /build/glibc-sMfBJT/glibc-2.31/csu/../csu/libc-start.c:308:16
    #20 0x4247ad in _start (/root/autodriver+0x4247ad)

AddressSanitizer can not provide additional info.
SUMMARY: AddressSanitizer: SEGV /root/duckdb/src/execution/radix_partitioned_hashtable.cpp:350:54 in duckdb::RadixPartitionedHashTable::GetData(duckdb::ExecutionContext&, duckdb::DataChunk&, duckdb::GlobalSinkState&, duckdb::GlobalSourceState&) const
==26117==ABORTING
```

**Comment by fuboat** [comment]
In fact, I'm not sure whether it is a input triggering bug, or an invalid SQL input.

I have found many inputs causing duckdb binary crashes by fuzzing. I want to know how to distinguish if an input triggers bug.

**Comment by Mytherin** [comment]
Thanks for the report! I have pushed a fix in #3355.

In principle, anything that triggers an assertion, an internal exception or an address sanitizer failure is considered a bug.

Which fuzzing software are you running against DuckDB?

**Comment by fuboat** [comment]
You can try [AFLplusplus](https://github.com/AFLplusplus/AFLplusplus). I configure AFL++ as follows:
- seeds_dir: the testcases in duckdb/test/*  (it is recommended to prune them to SQL only).
- fuzz target: the duckdb shell binary compiled by afl-clang-fast++.

I am also trying [squirrel](https://github.com/s3team/Squirrel) and my unpublished technology on duckdb. I will report the bugs found by these tools through github issue.
